### PR TITLE
Clarify rg-name and rg-region are optional for EE

### PR DESCRIPTION
**Pull request by ddebroy**
### Proposed changes

When trying to deploy Docker for Azure EE from the Azure marketplace, the Azure portal does not present the option to deploy to an existing resource group. So make corresponding changes to recommendations for creating the Azure Service Principal to address this deficiency. 

### Related issues (optional)

https://github.com/docker/orca/issues/5421

### CODE REVIEW

**Comment by mstanleyjones** [comment]
cc/ @friism @FrenchBen @nathanleclaire

File: docker-for-azure/index.md [hunk 1]
```diff
@@ -68,33 +68,54 @@ The number of workers you want in your swarm (1-100).
...
-
-If the script fails, it's typically because your Azure user account doesn't have sufficient privileges. Contact your Azure administrator.
-
-When setting up the ARM template, you will be prompted for the App ID (a UUID) and the app secret. If you are deploying Docker Community Edition for Azure and specified the resource group name and location parameters, please choose the option to deploy the template into an existing resource group and pass the same name and region/location that were passed above to create-sp-azure.
+A [Service Principal](https://azure.microsoft.com/en-us/documentation/articles/active-directory-application-objects/)
+is required to set up Docker for Azure. The Service Principal invokes Azure APIs as you scale the number of nodes up
```
Comment (ddebroy) [inline docker-for-azure/index.md]: Please reword `The Service Principal invokes ...` to `The Service Principal is used to invoke ...`

File: docker-for-azure/index.md [hunk 2]
```diff
@@ -68,33 +68,54 @@ The number of workers you want in your swarm (1-100).
...
-
-While `rg-name` and `rg-region` are optional, it's highly recommended that you create the resource group up front and scope the service principal to that specific resource group.
-
-If the script fails, it's typically because your Azure user account doesn't have sufficient privileges. Contact your Azure administrator.
-
-When setting up the ARM template, you will be prompted for the App ID (a UUID) and the app secret. If you specified the resource group name and location parameters, please choose the option to deploy the template into an existing resource group and pass the same name and region/location that were passed above to create-sp-azure.
+A [Service Principal](https://azure.microsoft.com/en-us/documentation/articles/active-directory-application-objects/)
+is required to set up Docker for Azure. The Service Principal is used to invoke Azure APIs as you scale the number of nodes up
+and down or deploy apps on your swarm that require configuration of the Azure Load Balancer. Docker provides a
+containerized helper-script called `docker4x/create-sp-azure` to help you create the Service Principal.
```
Comment (nathanleclaire) [inline docker-for-azure/index.md]: I'd say "helper script" (no hyphen)

File: docker-for-azure/index.md [hunk 3]
```diff
@@ -68,33 +68,54 @@ The number of workers you want in your swarm (1-100).
...
+1.  Download the latest version of `docker4x/create-sp-azure` to your local environment:
+   
+   ```bash
+   docker pull docker4x/create-sp-azure:latest
+   ```
+
+2. Run the `sp-azure` script with the following arguments:
+
+   ```bash
+   $ docker run -ti docker4x/create-sp-azure sp-name [rg-name rg-region]
```
Comment (nathanleclaire) [inline docker-for-azure/index.md]: Is there a particular reason why the `sp-name` is not in the brackets here?

**Comment by nathanleclaire** [comment]
Looks great just a couple of nitpicks

File: docker-for-azure/index.md (hunk 3 above)
Comment (ddebroy) [inline docker-for-azure/index.md]: `sp-name` is a required parameter. `rg-name` and `rg-region` are optional parameters.

Comment (mstanleyjones) [inline docker-for-azure/index.md]: Is it required? I think brackets imply a value that isn't required.

Comment (nathanleclaire) [inline docker-for-azure/index.md]: OK cool I thought it might be optional parameters but couldn't find any reference to that  nearby.

Comment (ddebroy) [inline docker-for-azure/index.md]: `rg-name` and `rg-region` are optional parameters and not necessary for any scenario. Therefore I put them in brackets. They are recommended however in the CE scenarios which is what we explain down below.

**Comment by mstanleyjones** [comment]
@johndmulhausen can you merge this or kick Netlify?

**Comment by johndmulhausen** [comment]
Sorry Netlify is blocking up the works. (kicks system)

**Pull request merged**